SharpRepository.MongoDbRepository/MongoDbRepositoryBase.cs

```diff
@@ -107,11 +108,17 @@ private void Initialize(IMongoDatabase mongoDatabase = null)
                         }
                     );
                 }
+
+            var collectionNameAttributes = EntityType.GetTypeInfo().GetOneAttribute<MongoDbCollectionNameAttribute>(inherit: true);
```

Comment:
This must be before `if (BsonClassMap.IsClassMapRegistered(typeof(T)))` in line 83!

If not after first map registration _collectionName will be null and throw exception